### PR TITLE
Added listener on mouseup to body for region usability improvement.

### PR DESCRIPTION
**Pull request by RocAlayo**
With this code i easier  to use the dragging and resizing of region because if the mouseup event happens outside the wavesurfer element this actions are stopped and you don't get stuck with it until you click again. This way is easier to adjust a region to the star or end of the wave.

### CODE REVIEW

File: plugin/wavesurfer.regions.js [hunk 1]
```diff
@@ -331,6 +331,9 @@ WaveSurfer.Region = {
             my.element.addEventListener('mousedown', onDown);
             my.wrapper.addEventListener('mouseup', onUp);
             my.wrapper.addEventListener('mousemove', onMove);
+            document.querySelector("body").addEventListener("mouseup", function () {
```
Comment (katspaugh) [inline plugin/wavesurfer.regions.js]: `document.body.addEventListener('mouseup', onUp)` — shorter, the same meaning.

Please, also use single quotes.

File: plugin/wavesurfer.regions.js [hunk 2]
```diff
@@ -331,6 +331,9 @@ WaveSurfer.Region = {
             my.element.addEventListener('mousedown', onDown);
             my.wrapper.addEventListener('mouseup', onUp);
```
Comment (katspaugh) [inline plugin/wavesurfer.regions.js]: This can be removed, as we don't need two identical events.

File: plugin/wavesurfer.regions.js (hunk 1 above)
Comment (katspaugh) [inline plugin/wavesurfer.regions.js]: The event handler on body should be removed on destroy. Subscribe to the wavesurfer's destroy event.

Comment (RocAlayo) [inline plugin/wavesurfer.regions.js]: In function `onUp` you do `e.stopPropagation();` and `e.preventDefault();`. This applied to the body can make some functionality of the web using this library to brake, that's why I put it this way. But If we remove `my.wrapper.addEventListener('mouseup', onUp);` we can delete the two lines that stop event bubbling.

**Comment by RocAlayo** [comment]
Ok, I have update the pull request and I put single quotes too. In the last update I forgot to change the single quotes :P

**Comment by katspaugh** [comment]
Is it working as before without `preventDefault`? Most importantly, no text selection should occur.

And you still aren't unsubscribing mouseup on wavesurfer destroy. If you add this last bit, we can merge it.

**Comment by RocAlayo** [comment]
I'm sorry maybe I'm undestanding wrong the code but nothing in regions plugins is subscribed to wavesurfer destroy. All is subscribed to remove of a region witch now also remove the event listener mouseup of body. Am I wrong?

**Comment by katspaugh** [comment]
We need to unsubscribe both on region remove and on global destroy. Previously, nothing was unsubscribed on destroy, because the subscribed DOM elements themselves were removed from DOM on destroy. Since `document.body` isn't going to be removed, it needs to be unsubscribed.

**Comment by RocAlayo** [comment]
Ok, got it, thanks for the explanation. I will change this and test if everything goes well without the `preventDefault` and then update the PR.

**Comment by katspaugh** [comment]
I guess it's working OK without `preventDefault`. Thanks!

**Pull request merged**